arduino/WandCore/TouchSurfaceClient.cpp

```diff
@@ -0,0 +1,115 @@
+/*
+  Capacitive Touch Surface Client
+
+  Manages a Touch Surface Input
+*/
+
+#include "TouchSurfaceClient.h"
+
+// Pin the touch surface is attached to
+int touchPin;
+
+// Touch surface constants
+const int touchThold = 50;                  // readings above mean the sensor is not being touched
+const int debounce = 20;                    // ms debounce period to prevent flickering when touching or stopping to touch the surface
+const int doubleTapGap = 250;               // max ms between taps for a double tap event
+const int holdTime = 1000;                  // ms hold period: how long to wait for hold event; TODO experiment if this can shorten
+
+// Touch surface state variables
+boolean isTouched = false;                  // whether or not the touch surface is being touched during this loop
+boolean wasTouchedLast = false;             // whether or not the touch surface was touched last loop
+boolean doubleTapWaiting = false;           // true if a double tap may be pending
+boolean doubleTapOnUp = false;              // true if double tap conditions have been met but the second tap is still being touched
+unsigned long downTime = -1;                // most recent time when the surface was first touched

```

Comment:
> // most recent time when the surface was first touched

I think _"first"_ in this comment is not correct. It looks like you still update `downTime` on the second touch of a double tap.

Reply:
yeah first isn't the right word here, I intended it to mean "started to be touched", not relating at all to single or double tap. Will tweak.